core/src/main/scala/com/gu/play/secretrotation/RotatingSecretComponents.scala

```diff
@@ -13,8 +13,11 @@ trait RotatingSecretComponents extends BuiltInComponentsFromContext {
 
   val secretStateSupplier: SnapshotProvider
 
-  override def configuration: Configuration = super.configuration ++
-    Configuration("play.http.secret.key" -> secretStateSupplier.snapshot().secrets.active)
```

Comment:
So `secretStateSupplier.snapshot().secrets.active` was not actually a `String`, it was a `SecretConfiguration` - trying to stuff that into string-based config did not go well:

```
play.api.UnexpectedException: Unexpected exception[BugOrBroken: bug in method caller: not valid to create ConfigValue from: SecretConfiguration(MySecretDoesNotActuallyLookLikeThis,None)]
        at play.core.server.DevServerStart$$anon$1.reload(DevServerStart.scala:186)
        at play.core.server.DevServerStart$$anon$1.get(DevServerStart.scala:124)
        at play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:222)
        at play.core.server.AkkaHttpServer.$anonfun$createServerBinding$1(AkkaHttpServer.scala:137)
        at akka.stream.impl.fusing.MapAsync$$anon$25.onPush(Ops.scala:1194)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:519)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:482)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:378)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:585)
        at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:469)
Caused by: com.typesafe.config.ConfigException$BugOrBroken: bug in method caller: not valid to create ConfigValue from: SecretConfiguration(MySecretDoesNotActuallyLookLikeThis,None)
        at com.typesafe.config.impl.ConfigImpl.fromAnyRef(ConfigImpl.java:281)
        at com.typesafe.config.impl.PropertiesParser.fromPathMap(PropertiesParser.java:165)
        at com.typesafe.config.impl.PropertiesParser.fromPathMap(PropertiesParser.java:95)
        at com.typesafe.config.impl.ConfigImpl.fromAnyRef(ConfigImpl.java:264)
        at com.typesafe.config.impl.ConfigImpl.fromPathMap(ConfigImpl.java:200)
        at com.typesafe.config.ConfigFactory.parseMap(ConfigFactory.java:1039)
        at com.typesafe.config.ConfigFactory.parseMap(ConfigFactory.java:1050)
        at play.api.Configuration$.from(Configuration.scala:139)
        at play.api.Configuration$.apply(Configuration.scala:145)
        at com.gu.play.secretrotation.RotatingSecretComponents.configuration(RotatingSecretComponents.scala:17)
```